app/backend/application/database.py

```diff
@@ -20,11 +22,16 @@ def delete_record(record_id: str):
     db.session.commit()
 
 
-def set_record_uses(record_id: str, max_uses: int):
+def set_record_uses_expiry(record_id: str, max_uses: int, expiry_hours: int, expiry_minutes: int):
     record = db.session.query(Sessions).filter_by(id=record_id).first()
     if record is None:
         raise werkzeug.exceptions.BadRequest("Incorrect ID")
+    if max_uses == 0:
```

Comment:
If you introduce smth like `str_to_nonnegative_int` and `str_to_positive_int`, you won't need extra validation steps outside of conversion functions.